<?xml version="1.0"?>
<launch>
   <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
<!--     <param name="width" type="int" value="1280" />
    <param name="height" type="int" value="960" /> -->
    <param name="width" type="int" value="640" />
    <param name="height" type="int" value="480" />
    <param name="fps" type="int" value="120" />
    <param name="frame" type="string" value="wide_stereo" />
    <param name="auto_focus" type="bool" value="False" />
    <param name="focus_absolute" type="int" value="0" />

    <param name="device" type="string" value="/dev/video0" />

    <param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example.yaml" />
  </node>

  <!-- BORDER CUTTER -->
  <node pkg="res_rob_vision" type="area_borders_node" name="area_borders" output="screen">
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="area_borders/image_processed" />

        <param name="DEBUGGING" value="true" />

        <param name="color_yellow" value="true" />
        <param name="color_orange" value="true" />
        <param name="color_red" value="true" />

        <param name="HL_threshold" value="100" />
        <param name="HL_minLineLength" value="100" />
        <param name="HL_maxLineGap" value="50" />

  </node>

  <!-- MOTION DETECTION -->
<!--   <node pkg="motion_detection" type="motion_detection_node" name="motion_detection" output="screen">
        <param name="sub_topic" value="area_borders/image_processed" />
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="motion_detection/info" />

        <param name="DEBUGGING" value="false" />
        <param name="erode" value="true" />
        <param name="erosion_kernel_size" value="3" />
        <param name="threshold_value" value="10" />
        <param name="threshold_method" value="0" /> 
  </node> -->
  <!-- 0-SIMPLE 1-ADAPTIVE_GAUSSIAN 2-ADAPTIVE_MEAN 3-OTSU -->


  <!-- QRCODE READER -->
<!--   <node pkg="qr_reader" type="qr_reader_node" name="qr_reader" output="screen">
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="qr_reader/qr_message" />
        <param name="points_pub_topic" value="qr_reader/qr_points" />

        <param name="DEBUGGING" value="false" />
        <param name="qr_square_width" value="182" />
        
  </node> -->

  <!-- HZL READER -->
<!--   <node pkg="hzl_reader" type="hzl_reader_node" name="hzl_reader" output="screen">
        <rosparam command="load" file="/home/nikos/polymechanon_vision_ws/src/hzl_reader/include/Hazardous_labels/labels.yaml" />
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="hzl_reader/info" />

        <param name="DEBUGGING" value="false" />
        <param name="DBG_MESSAGES" value="false" />
        <param name="contours_of_interest_area" value="10000" /> 
        <param name="match_method" value="4" />      
        <param name="matching_square_size" value="150" />
        <param name="gray_match" value="true" />
        <param name="canny_par_1" value="100" />   
        <param name="canny_par_2" value="200" />   
        <param name="canny_par_3" value="3" />   
        <param name="contrast" value="1.0" /> 
        <param name="brightness" value="1" />   

  </node> -->

  <!-- FACE DETECTOR -->
<!--     <node pkg="face_detection" type="face_detection_node" name="face_detection" output="screen">
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="face_detection/info" />
        <param name="DEBUGGING" value="false" /> 
        <param name="min_face_size" value="200" />
        <param name="max_face_size" value="500" />     
  </node> -->

  <!-- CIRCLE DETECTOR -->
<!--     <node pkg="res_rob_vision" type="circle_detection_node" name="circle_detection" output="screen">
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="circle_detection/info" />
        <param name="DEBUGGING" value="false" />      
        <param name="method" value="3" /> 
        <param name="threshold_value" value="10" />
        <param name="erode_dilate_kernel_size" value="3" />
        <param name="HC_minDist" value="2" />
        <param name="HC_param1" value="200" />
        <param name="HC_param2" value="90" />
        <param name="HC_minRadius" value="10" />
        <param name="HC_maxRadius" value="2" />
  </node> -->

   <!-- VIDEO TESTS -->
<!--   <node pkg="video_tests" type="video_tests_node" name="video_tests" output="screen">
        <param name="sub_topic" value="camera/image_raw" />
        <param name="pub_topic" value="video_tests/info" />

        <param name="DEBUGGING" value="true" />
        <param name="erode" value="true" />
        <param name="erosion_kernel_size" value="3" />
        <param name="threshold_value" value="10" />
        <param name="threshold_method" value="0" /> 
        <param name="contrast" value="1.0" /> 
        <param name="brightness" value="1" />   
  </node> -->

  <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui" output="screen" /> 
  
</launch>
